test/extended/util/image/README.md

```diff
@@ -0,0 +1,53 @@
+# Images used by e2e tests
+
+We limit the set of images used by e2e to reduce duplication and to allow us to provide offline mirroring of images for customers and restricted test environments. Every image used in e2e must be part of this utility package or referenced by the upstream `k8s.io/kubernetes/test/utils/image` package.
+
+All images used by e2e are mirrored to:
+
+  quay.io/openshift/community-e2e-images:TAG
+
+for easy mirroring by the `openshift-tests images` command.
+
+## To add a new image:
+
+1. Identify whether your use case can be solved by an existing image:
+   * The standard "shell" image provided by `ShellImage()` that is available on all clusters and has bash + a set of standard tools available. Use this by default.
+   * An upstream image that is provided by `test/utils/image`.
+2. If your use case is novel, you must:
+   * Describe your use case in a PR to this file and have it approved
+   * Define a standard CI build and publish the image to quay.io in the openshift namespace
+   * Add the new reference to the `init()` method in this package
+   * Have the automation promote the image to the quay mirror location.
+
+When adding a new image, first make the code changes and compile the `openshift-tests` binary. Then run `hack/update-generated-bindata.sh` to update `test/extended/util/image/zz_generated.txt`. Contact one of the OWNERS of this directory and have them review the image for inclusion into our suite (usually granted in the process above). Before merge and after review they will run the following command to mirror the content to quay:
+
+    openshift-tests images --upstream --to-repository quay.io/openshift/community-e2e-images | oc image mirror -f - --filter-by-os=.*
```

Comment:
why is this step needed when the above list of steps includes "have automation promote the image to the quay mirror location"?

alternatively, why isn't this command just run on a schedule?  Worst case you'd have to first merge your addition of the new image, wait for the mirror to happen, and then merge your test changes that depend on the new image.

Reply:
Because we don't want to mirror things that haven't been approved.  Future steps include rebuilding all these images ourselves instead of taking random binaries from upstream.